src/applications/coronavirus-vaccination/hooks/useRecaptcha.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useCallback } from 'react';
+
+// temporary localhost key. need to use real key.  Can this remain public or should still be stored in credStash?
+const SITE_KEY = '6Lcbbf0ZAAAAABAWyrsMSyTd1RRAM71rrk350SLa';
```

Comment:
I think this is okay to be public per https://developers.google.com/recaptcha/intro 

> The site key is used to invoke reCAPTCHA service on your site or mobile application.

So yeah, I _think_ we're fine to store an actual site key on GitHub (that's what we did with the MapBox API's site key IIRC) but we'll have to have the secret key in Credstash. 

> The secret key authorizes communication between your application backend and the reCAPTCHA server to verify the user's response. The secret key needs to be kept safe for security purposes.

If I follow this Captcha flow at a high-level....That secret key would need to be in Credstash, made available in the [API settings](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/vets-api/prod-settings.local.yml.j2), then when the form is submitted, the API can access that secret key and do the verification (via the token) before actually submitting it. If the verification fails, the API could return a special status code indicating to the FE that.

Does that all sound correct?

Reply:
Yep that sounds right.  I am also thinking it might be better to have a separate call to BE just for recaptcha before the user submits so we can, based on the returned 'score', opt to show the users a V2 style reCaptcha so they can prove they are human.  I am still digging in on how best to implement that part of it. Some examples I saw had the Captcha token added to an existing payload but I am not sure if that always makes sense.